base/reducedim.jl

```diff
@@ -322,7 +322,7 @@ julia> mapreduce(isodd, |, a, dims=1)
 mapreduce(f, op, A::AbstractArrayOrBroadcasted; dims=:, init=_InitialValue()) =
     _mapreduce_dim(f, op, init, A, dims)
 mapreduce(f, op, A::AbstractArrayOrBroadcasted...; kw...) =
-    reduce(op, map(f, A...); kw...)
+    reduce(op, Generator(f, A...); kw...)
```

Comment:
This broke the case where both `dims` and `init` kwargs are specified. Can't we just do
```suggestion
    mapreduce(Base.splat(f), op, zip(A...); kw...)
```
here directly instead?